doc/API.md

```diff
@@ -146,3 +153,147 @@ Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 409, errno 409: Already exists.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /
+Get the list of all registered users.
+
+### Request
+```ssh
+GET / HTTP/1.1
+```
+Requests must include a authorization token containing a [bearer token](#authentication).
+
+### Response
+Successful requests will produce a "200 OK" response with a body containing the list of all registered users:
+```ssh
+HTTP/1.1 200 OK
+Connection: close
+[{
+  "id": "hfkjsIklksadhs",
+  "username": "admin"
+  "email": "admin@domain.org"
+}, {
+  "id": "InR5cCI6IkpXVCJ",
+  "username": "pepe"
+  "email": "user@domain.org"
+}]
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /:id
+Start user activation.
+
+### Request
+```ssh
+GET /InR5cCI6IkpXVCJ HTTP/1.1
+```
+
+### Response
+Successful requests will produce a "200 OK" response with a body containing an activation token with a short ttl and the inactive user email:
+```ssh
+HTTP/1.1 200 OK
+Connection: close
+{
+  "activation_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjEiLCJuYW1lIjoidXNlcm5hbWUifQ.IEMuCIdMp53kiUUoBhrxv1GAPQn2L5cqhxNmCc9f_gc"
+  "email": "user@domain.org"
+}
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 410, errno 410: Gone. The user does not exist or was already activated.
+
+## PUT /:id
+Edit user information.
+
+### Request
+Requests must include a authorization token containing a [bearer token](#authentication). This token can be an activation token with a short ttl or a session token with a longer ttl.
+___Parameters___
+* username - User name.
+* password - User password.
+
+```ssh
+PUT /:id HTTP/1.1
+Content-Type: application/json
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+{
+  "username": "pepe",
+  "password": "whatever"
+}
+```
+
+### Response
+Successful requests will produce a "204 No Content" response for requests authenticated with a session token or a "200 OK" response for requests authenticated with an activation token.
+```ssh
+HTTP/1.1 200 OK
+Connection: close
+{
+  "session_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjEiLCJuYW1lIjoidXNlcm5hbWUifQ.IEMuCIdMp53kiUUoBhrxv1GAPQn2L5cqhxNmCc9f_gc"
+}
+```
+
+or
+
+```ssh
+HTTP/1.1 204 No Content
+Connection: Cloe
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 100: Invalid user name. Missing or malformed user name.
+* status code 400, errno 102: Invalid password. The password should have a minimum of 8 chars.
+* status code 400, errno 400: Bad request.
+* status code 409, errno 409: Conflict. The user name is already registered.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## DELETE /:id
+Delete a user.
+
+### Request
```

Comment:
I guess I will need to read the implementation, but can any user remove any other user. Or better said, will be all user created admins?